types/consensus_data.go

```diff
@@ -251,74 +245,180 @@ func (cd *ValidatorConsensusData) GetBlockData() (blk *api.VersionedProposal, si
+	if len(a.Aggregators) != len(a.Attestations) {
+		return nil, nil, NewError(AggCommAggAttCntMismatchErrorCode, "aggregators and attestations count mismatch")
+	}
+
+	proofs := make([]*spec.VersionedAggregateAndProof, 0, len(a.Aggregators))
+	hashRoots := make([]ssz.HashRoot, 0, len(a.Aggregators))
+
+	for i, aggregator := range a.Aggregators {
+		// Decode attestation based on version
+		var aggregateAndProof *spec.VersionedAggregateAndProof
+
+		switch a.Version {
+		case spec.DataVersionPhase0, spec.DataVersionAltair, spec.DataVersionBellatrix, spec.DataVersionCapella, spec.DataVersionDeneb:
+			agg := &phase0.AggregateAndProof{
+				AggregatorIndex: aggregator.ValidatorIndex,
+				SelectionProof:  aggregator.SelectionProof,
+			}
+			// Unmarshal the attestation
+			att := &phase0.Attestation{}
+			if err := att.UnmarshalSSZ(a.Attestations[i]); err != nil {
+				return nil, nil, WrapError(UnmarshalSSZErrorCode, fmt.Errorf("failed to unmarshal attestation: %w", err))
+			}
+			agg.Aggregate = att
+
+			aggregateAndProof = &spec.VersionedAggregateAndProof{
+				Version: a.Version,
+			}
+			// Set the appropriate version field and store hash root
+			switch a.Version {
+			case spec.DataVersionPhase0:
+				aggregateAndProof.Phase0 = agg
+				hashRoots = append(hashRoots, agg)
+			case spec.DataVersionAltair:
+				aggregateAndProof.Altair = agg
+				hashRoots = append(hashRoots, agg)
+			case spec.DataVersionBellatrix:
+				aggregateAndProof.Bellatrix = agg
+				hashRoots = append(hashRoots, agg)
+			case spec.DataVersionCapella:
+				aggregateAndProof.Capella = agg
+				hashRoots = append(hashRoots, agg)
+			case spec.DataVersionDeneb:
+				aggregateAndProof.Deneb = agg
+				hashRoots = append(hashRoots, agg)
+			default:
+				panic("unhandled default case")
+			}
+
+		case spec.DataVersionElectra, spec.DataVersionFulu:
+			agg := &electra.AggregateAndProof{
+				AggregatorIndex: aggregator.ValidatorIndex,
+				SelectionProof:  aggregator.SelectionProof,
+			}
+			// Unmarshal the attestation
+			att := &electra.Attestation{}
+			if err := att.UnmarshalSSZ(a.Attestations[i]); err != nil {
+				return nil, nil, WrapError(UnmarshalSSZErrorCode, fmt.Errorf("failed to unmarshal electra attestation: %w", err))
+			}
+			agg.Aggregate = att
+
+			aggregateAndProof = &spec.VersionedAggregateAndProof{
+				Version: a.Version,
+			}
+
+			switch a.Version {
+			case spec.DataVersionElectra:
+				aggregateAndProof.Electra = agg
+			case spec.DataVersionFulu:
+				aggregateAndProof.Fulu = agg
+			default:
+				panic("unhandled default case")
+			}
+			hashRoots = append(hashRoots, agg)
+
+		default:
+			return nil, nil, WrapError(UnknownBlockVersionErrorCode, fmt.Errorf("unsupported version %s", a.Version.String()))
 		}
 
-		return &spec.VersionedAggregateAndProof{Version: cd.Version, Fulu: ret}, ret, nil
-	default:
-		return nil, nil, fmt.Errorf("unknown aggregate and proof version %d", cd.Version)
+		proofs = append(proofs, aggregateAndProof)
 	}
```

Comment:
this is really confusing
`proofs` are a VersionedAggregateAndProofs
and `hashRoots` are the same in SSZ form?

this is really confusing. The names of the variables are bad. Also you can also extract the `hashroot` easily from the `proof`.

Maybe just return `aggregateAndProofs` list and that's it?